### PR TITLE
Make interactive mode strategy-aware

### PR DESCRIPTION
**Pull request by george-elphick-talieisin**
## Summary

- Interactive mode (`--interactive`) now respects file-type strategies — it only suggests changes in regions the strategy considers safe (comments in code, prose in markdown, etc.)
- Previously it bypassed strategies entirely, suggesting changes inside code strings, markdown code blocks, CSS properties, etc.
- JSON files fall back to non-interactive processing with a stderr warning, since `JSONStrategy` reformats content via `json.dumps`

## Implementation

- Added `find_safe_replacements()` to `FileProcessingStrategy` base class — runs `process()` once, then uses positional comparison to determine which candidates were actually applied
- Added `supports_interactive` flag to the strategy API (`False` for `JSONStrategy`) so the contract is enforced rather than relying on `isinstance` checks
- Fixed a correctness bug in the positional algorithm where shorter replacements (e.g. `appall`→`appal`) could falsely match in excluded regions

## Test plan

- [x] `TestFindSafeReplacements` — per-strategy filtering (Code, Markdown, CSS, HTML, PlainText) including shorter-replacement edge cases
- [x] `TestInteractiveStrategyInvariant` — parameterised: approve-all via `find_safe_replacements` + `apply_replacements` equals `strategy.process()` for all non-JSON strategies
- [x] `TestJsonInteractiveFallback` — verifies `supports_interactive` flag
- [x] All 148 tests pass

## Related issues

- #7 — pre-existing: JSONStrategy falls back to raw `correct_text()` on malformed JSON
- #8 — pre-existing: interactive word grouping is O(n²) and duplicated

### CODE REVIEW

File: britfix.py [hunk 1]
```diff
@@ -115,13 +116,13 @@ def create_backup(filepath: str) -> str:
...
     with open(filepath, 'r', encoding='utf-8') as f:
         content = f.read()
-    
-    # Find all potential replacements
-    replacements = corrector.find_replacements(content)
+
+    # Find replacements filtered by strategy segmentation rules
+    replacements = strategy.find_safe_replacements(content, corrector)
     if not replacements:
         return content, {}
```
[the remark this answers is not in the document]
Comment (george-elphick-talieisin) [reply britfix.py:128]: Already tracked in #8 — pre-existing issue, out of scope for this PR.

File: britfix_core.py [hunk 2]
```diff
@@ -161,11 +161,45 @@ def load_spelling_mappings(file_path: Optional[str] = None) -> Dict[str, str]:
...
+            # Check for original text first — strategy left it unchanged
+            if corrected[adj_start:adj_end_original] == original:
+                # Not applied by strategy, no offset change
+                continue
+            # Strategy changed this region — verify it's the expected replacement
+            adj_end_replaced = adj_start + len(replacement)
+            if corrected[adj_start:adj_end_replaced] == replacement:
+                safe.append((start, end, original, replacement))
+                offset += len(replacement) - len(original)
+            # else: unexpected state, skip conservatively
```
[the remark this answers is not in the document]
Comment (george-elphick-talieisin) [reply britfix_core.py]: Good catch — confirmed the false negative with `catalog`→`catalogue` (and 22 other mappings). Fixed in b89583f using the boundary-character approach you suggested: check the replacement slice first, verify the following character matches the post-word boundary from the source content. This handles both directions (shorter like `appall`→`appal` and longer like `catalog`→`catalogue`).

File: test_britfix.py [hunk 3]
```diff
@@ -1283,5 +1285,115 @@ def test_windows(self, monkeypatch):
...
+
+    def test_code_strategy_shorter_replacement_distill(self, corrector):
+        """CodeStrategy: distill->distil only in comment, not string."""
+        code = '# distill\nname = "distill"\n'
+        strategy = CodeStrategy()
+        safe = strategy.find_safe_replacements(code, corrector)
+        assert len(safe) == 1
+        assert safe[0][2] == 'distill'
+
+
```
[the remark this answers is not in the document]
Comment (george-elphick-talieisin) [reply test_britfix.py:1376]: Added in b89583f — tests for `catalog`→`catalogue` and `program`→`programme`, plus a parameterised invariant case.
